model/src/main/kotlin/PackageCuration.kt

```diff
@@ -59,9 +61,10 @@ data class PackageCuration(
      */
     private fun isApplicableIvyVersion(pkgId: Identifier) =
         try {
-            val pkgIvyVersion = Semver(pkgId.version, Semver.SemverType.IVY)
-            pkgIvyVersion.satisfies(id.version)
+            val requirement = Requirement.buildIvy(id.version)
+            requirement.isSatisfiedBy(pkgId.version)
         } catch (e: SemverException) {
+            log.warn { "Curation version '${id.version}' does not parse as an Ivy version." }
```

Comment:
We should consider throwing the exception if it occurs in `buildIvy`, because that would mean that the curations are broken.

Reply:
| .. because that would mean that the curations are broken.

I believe that an exception in `buildIvy` does not apply that the curations are broken.
IIRC there are some limitations in the library, but I don't recall exactly which one that was.
Maybe for any of the the exception is thrown? `1.2.3-release` or `1.2.3.4`.

Putting it more general: Are curation versions in ORT limited to Ivy version and or Ivy versions the library can handle? If that's not the case a warning may be questionable.

Reply:
> I believe that an exception in `buildIvy` does not apply that the curations are broken.

Correct, curation versions may be non-Ivy versions, like the examples you gave. But we already failed the plain string comparison when `isApplicableIvyVersion()` is called, and Ivy version are the only "wildcard" version we support, so it might indeed be feasible here to not catch the exception but add it as an error to the analyzer result be the caller.

Reply:
Ah, crap, no, we really can't (re-)throw that exception here. If there is a curation for `Maven:group:name:1-foo` and the package is `Maven:group:name:1-bar`, then the code at

https://github.com/oss-review-toolkit/ort/pull/3410/files#diff-5b02296a9d7fa7c14ae97135a6fab1a33fc1c0d14028b126420a512655cc88d5R76

would call `isApplicableIvyVersion()` because `id.version.equalsOrIsBlank()` failed, and `isApplicableIvyVersion()` would throw, although it's a valid curation.